packages/xo-web/src/xo-app/sr/tab-disks.js

```diff
@@ -319,6 +337,98 @@ export default class SrDisks extends Component {
     vbdsByVdi => mapValues(vbdsByVdi, vbds => some(vbds, 'attached'))
   )
 
+  _getRequiredHosts = createSelector(
+    () => this.props.allVdis,
+    () => this.props.vbds,
+    () => this.props.srs,
+    createCollectionWrapper(_ => _),
+    (vdis, vbds, srs, selectedVdis) => {
+      const vbdsByVm = groupBy(vbds, 'VM')
+      const requiredHosts = new Set()
+      const vms = new Set()
+
+      selectedVdis.forEach(vdi => {
+        // Get the required host for each VM contains this VDI
+        forEach(vdi.$VBDs, vbdId => {
+          const vm = get(() => vbds[vbdId].VM)
+          if (vm !== undefined && !vms.has(vm)) {
+            vms.add(vm)
+            forEach(vbdsByVm[vm], vbd => {
+              let sr
+              let vmVdi
+              if (
+                !vbd.is_cd_drive &&
+                (vmVdi = vdis[vbd.VDI]) !== undefined &&
+                (sr = srs[vmVdi.$SR]) !== undefined &&
+                !isSrShared(sr)
+              ) {
+                requiredHosts.add(sr.$container)
+                return false
+              }
+            })
+          }
+        })
+
+        if (requiredHosts.size > 1) {
+          // No need to get all the required hosts
+          return false
+        }
+      })
+
+      if (
+        vms.size === 1 &&
+        vbdsByVm[vms.values().next().value].filter(vbd => !vbd.is_cd_drive)
+          .length === selectedVdis.length
+      ) {
+        // No required host when the selected VDIs are all the VDIs attached to a VM
+        return
+      }
+
+      return requiredHosts
+    }
+  )
```

Comment:
So in `vm/disks` we don't check if migrating the disk will be an issue for any other VMs that have that disk attached to it?

Reply:
Yes, we need to fix it in another PR?

Reply:
What about this:
- In `vm/disks`:
  - we keep the current complex check for the VM we're looking at
  - we add an extra check in case there are 2 VBDs or more on the VDI to warn the user about breaking other VMs (but he may not have permissions on them so we won't be able to tell him if it will break them for sure or not)
- In `sr/disks`:
  - we only add a simple warning about the risk of breaking the VMs the VDI is attached to (and the user has to check by himself if it will break anything, otherwise I think it's getting too complicated to check all the permissions for everything). Let me know what you think but I don't think we can properly handle a case like this:
    - I have ACLs on SR1 and SR2 and that's it
    - SR1 is shared and SR2 is local to host2
    - I want to migrate a VDI from SR1 to SR2
    - the VDI is attached to a VM, that VM has another VDI sitting on SR3 which is local to host1